src/storage/src/source/reclock.rs

```diff
@@ -146,16 +144,13 @@ where
         &self,
         src_ts: &FromTime,
     ) -> Result<Antichain<IntoTime>, ReclockError<FromTime>> {
-        let inner = self.inner.borrow();
-        if !PartialOrder::less_than(&inner.as_of, &inner.upper) {
+        if !self.initialized() {
             return Err(ReclockError::Uninitialized);
         }
+        let inner = self.inner.borrow();
         if inner.source_upper.less_equal(src_ts) {
             return Err(ReclockError::BeyondUpper(src_ts.clone()));
         }
-        if !inner.source_since.less_equal(src_ts) {
-            return Err(ReclockError::NotBeyondSince(src_ts.clone()));
-        }
```

Comment:
I would say we need additional docs on `reclock_time` to explain this behavior, something like:

```
/// Note that its the _caller's responsibility_ to not attempt to reclock a `FromTime` that can no longer be reclocked
/// after compaction. In practice we do not expose any way to obtain a `FromTime` `since` frontier for the active
/// trace, but it will always be safe to reclock times that are _beyond_ the `source_upper_at_frontier` for the any
/// frontier we called `compact` with
///
/// This behavior may seem curious, but it preserves some internal logic; specifically, it preserves this equality:
/// ```
/// compact(reclock(collection, trace)) == reclock(collection, compact(trace))`
/// ```

Reply:
Also, we can no longer return a `ReclockError<FromTime>::NotBeyondSince` anywhere huh?

Reply:
I don't think this doc comment is correct. There is no `FromTime` timestamp that can no longer be reclocked. All timestamps are always reclockable, and the result of reclocking respects compaction. 

> Also, we can no longer return a ReclockError<FromTime>::NotBeyondSince anywhere huh?

The `source_upper_at_frontier` method still returns it. I had a branch somewhere that removed it from there too but I'll get to it at some later time.

Reply:
Hmm, you are right; I think its worth mentioning at least the equality that this behavior offers, in a comment